internal/x/exec/runner.go

```diff
@@ -811,7 +814,15 @@ func (r *runner) printFailures(filename string, meta *meta, failures ...*text.Fa
 			}
 		}
 		if shouldPrint {
-			if err := failure.Fprintln(bufWriter, failureFields...); err != nil {
+			if r.harbormasterOutput {
+				data, err := json.Marshal(phab.TextFailureToHarbormasterLintResult(failure))
+				if err != nil {
+					return err
```

Comment:
Should we add more context to the error here? It might help to include something about harbormaster so that it's not just the raw json marshal error.

Reply:
My general philosophy is to not over-complicate (in my opinion) errors here - this will virtually never actually produce an error as it is, and if it does, it’s a system bug that will be easy to find from the error message that is produced. Users can’t take any action on it.

Open to changing.

Reply:
Good point, I'm fine with leaving this as is. If it becomes an issue later on (doubtful it will), it's an easy change.